src/adapter/src/coord/sequencer/inner.rs

```
@@ -3068,6 +3073,155 @@ impl Coordinator {
+            .controller
+            .storage
+            .export(id)
+            .expect("sink known to exist")
+            .write_frontier;
+        let as_of = read_hold.least_valid_read();
+        assert!(write_frontier.iter().all(|t| as_of.less_than(t)));
+
+        let catalog_sink = Sink {
+            create_sql: sink.create_sql,
+            from: sink.from,
+            connection: sink.connection.clone(),
+            envelope: sink.envelope,
+            version: sink.version,
+            with_snapshot,
+            resolved_ids,
+            cluster_id: in_cluster,
+        };
+
+        let ops = vec![catalog::Op::UpdateItem {
+            id,
+            name: self.catalog.get_entry(&id).name().clone(),
+            to_item: CatalogItem::Sink(catalog_sink),
+        }];
+
+        self.catalog_transact(Some(session), ops).await?;
+
+        let status_id = Some(
+            self.catalog()
+                .resolve_builtin_storage_collection(&mz_catalog::builtin::MZ_SINK_STATUS_HISTORY),
+        );
+
+        let from_entry = self.catalog().get_entry(&sink.from);
+        let storage_sink_desc = StorageSinkDesc {
+            from: sink.from,
+            from_desc: from_entry
+                .desc_opt()
+                .expect("sinks can only be built on items with descs")
+                .into_owned(),
+            connection: sink
+                .connection
+                .clone()
+                .into_inline_connection(self.catalog().state()),
+            envelope: sink.envelope,
+            as_of,
+            with_snapshot,
+            version: sink.version,
+            status_id,
+            from_storage_metadata: (),
+        };
+
+        self.controller
+            .storage
+            .alter_export(
+                id,
+                ExportDescription {
+                    sink: storage_sink_desc,
+                    instance_id: in_cluster,
+                },
+            )
+            .await?;
```

I'm not sure it's safe to return an error here. Our historical rule has been: no errors returned after `catalog_transact`. Why does `alter_export` need to return an error? If it must return an error, what cleanup needs to happen in the catalog for next reboot to work?

---

I must have copy/pasted the wrong thing. We never expect this to fail so I will add an `.unwrap_or_terminate("cannot fail to alter source desc");` on it